src/components/legend.ts

```diff
@@ -167,26 +172,15 @@ export module Components {
 
     public _requestedSpace(offeredWidth: number, offeredHeight: number): _SpaceRequest {
       var estimatedLayout = this._calculateLayoutInfo(offeredWidth, offeredHeight);
-      var rowLengths = estimatedLayout.rows.map((row: string[]) => {
-        return d3.sum(row, (entry: string) => estimatedLayout.entryLengths.get(entry));
-      });
-
-      var longestRowLength = Utils.Methods.max(rowLengths, 0);
 
-      var longestUntruncatedEntryLength = Utils.Methods.max<string, number>(this._scale.domain(), (d: string) =>
-                                            this._measurer.measure(d).width, 0);
-      longestUntruncatedEntryLength += estimatedLayout.textHeight + this._padding;
-      var desiredWidth = this._padding + Math.max(longestRowLength, longestUntruncatedEntryLength);
+      var untruncatedRowLengths = estimatedLayout.rows.map((row: string[]) => {
+        return d3.sum(row, (entry: string) => estimatedLayout.untruncatedEntryLengths.get(entry));
+      });
```

Comment:
Probably too long to fit one one line. I think I can remove the inferred types though.